src/components/Integrations/Shopify/Shopify.js

```diff
@@ -99,6 +81,31 @@ const Shopify = ({ intl, dependencies: { shopifyClient } }) => {
     </>
   );
 
+  const getData = useCallback(async () => {
+    const result = await shopifyClient.getShopifyData();
+    if (!result.success) {
+      setError(<FormattedHTMLMessage id="validation_messages.error_unexpected_HTML" />);
+    } else if (result.value.length) {
+      setConnectionData({ shops: result.value, isConnected: true });
+      setIsConnecting(false);
+      setError(null);
```

Comment:
What do you think about convert _error_ and _is connecting_ states in properties like _shops_ and _is connected_?

Reply:
I've talk about this also with @fcoronelmakingsense this states do not update all at once. What difference this would make? This is to make code more readable and that's it?

Reply:
We've talked this over, I will be uploading a fix shortly.